airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -434,10 +434,16 @@ private List<UUID> updateOperations(final WebBackendConnectionUpdate webBackendC
       throws JsonValidationException, ConfigNotFoundException, IOException {
     final ConnectionRead connectionRead = connectionsHandler
         .getConnection(webBackendConnectionUpdate.getConnectionId());
-    final List<UUID> originalOperationIds = new ArrayList<>(connectionRead.getOperationIds());
+
+    final List<UUID> originalOperationIds =
+        connectionRead.getOperationIds() == null ? new ArrayList<>() : new ArrayList<>(connectionRead.getOperationIds()); // wrap to make modifiable
```

Comment:
by modifiable do you just mean that you're doing a defensive copy?

Reply:
Without creating a modifiable copy, we're unable to remove elements from the list here: https://github.com/airbytehq/airbyte/pull/16026/files/35e1c36a431cdec4ace463fa7b785a6eb6a170f9#diff-e581bd93015224d4dcccc5496889956d493634d619003e007dc893fe00a4cc9eR456

I was confused why the original code was making the copy, so I tried removing it but ran into an IllegalOperationException. So I brought it back with a comment, I'll tweak the comment to be a little more descriptive

Reply:
ah. makes sense.